cli/src/commands/extensions/api.rs

```diff
@@ -329,15 +330,17 @@ async fn get_package_details(
     let state = ExtensionState::from(op_state);
     let api = state.api().await?;
 
-    let package_type = PackageType::from_str(&package_type)
-        .map_err(|e| anyhow!("Unrecognized package type `{package_type}`: {e:?}"))?;
-    api.get_package_details(&PackageDescriptor {
+    api.submit_package(&PTPackageSpecifier {
         name: name.to_string(),
         version: version.to_string(),
-        package_type,
+        registry: package_type,
     })
     .await
     .map_err(Error::from)
+    .and_then(|resp| match resp {
+        PackageSubmitResponse::AlreadyProcessed(data) => Ok(data),
+        _ => Err(anyhow!("Package has not yet been processed.")),
```

Comment:
Why aren't we exposing all the enum variants to the user? Seems like useful information for an extension author.

Reply:
Maybe. But that's a breaking change to the extension API. Unless you just mean using different error messages for `New` vs `AlreadySubmitted`